### PR TITLE
Friendship V1

### PR DESCRIPTION
**Pull request by wathigo**
Tasks Description:
1. Creation of friendship models.
2. User models methods implementation. They include:
    - Get all the users friends
    - Get mutual friends between current_user and another user.
    - Get pending request.
3. Implementation of views for friendship feature.
Please use these credentials to see a working notification feature 
email: `example-1@facebook.org`
password: `foobar`

### CODE REVIEW

**Comment by mcrd25** [comment]
# Currently under review

File: fb/app/models/friendship.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class Friendship < ApplicationRecord
+  belongs_to :user
+  belongs_to :friend, class_name: 'User'
+  validates :user_id, presence: true
```
Comment (mcrd25) [inline fb/app/models/friendship.rb]: [OPTIONAL]
This actually is not necessary.

File: fb/app/models/friendship.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  validates :friend_id, presence: true
```
Comment (mcrd25) [inline fb/app/models/friendship.rb]: Same

File: fb/app/models/friendship.rb (hunk 1 above)
Comment (wathigo) [reply fb/app/models/friendship.rb]: Sure, I will remove this

File: fb/app/models/friendship.rb (hunk 2 above)
Comment (wathigo) [reply fb/app/models/friendship.rb]: Sure

**Comment by wathigo** [comment]
Thank you for the review. Am going to fix the notification feature.

**Pull request merged**